### DISCUSSION TITLE
Error with AuxKernal: Invalid function, Syntax error: Unknown identifier

### DISCUSSION THREAD

**Discussion by wlindqwister** [idaholab/moose#28157] [Q&A Getting Started]
### Check these boxes if you have followed the posting rules.

- [X] I have consulted the [Posting Guidelines](https://github.com/idaholab/moose/discussions/18270).
- [X] I have searched the [Discussions Forum](https://github.com/idaholab/moose/discussions) and [MOOSE Framework Troubleshooting](https://mooseframework.inl.gov/help/troubleshooting.html) and have not found what I was looking for
- [X] Q&A Getting Started is the most appropriate category for my question (trouble installing, beginner user, ...)

### Issue or question about MOOSE

Hello everyone, I'm relatively new with MOOSE and am trying to calculate an auxiliary variable 'density' based off of nodal distance from a boundary using the the `NearestNodeDistanceAux` aux kernal paired with `ParsedAux` (one step beyond what is demonstrated [here](https://github.com/idaholab/moose/pull/3966). I keep running into the following error when trying to calculate:

"The following error occurred in the AuxKernel 'density' of type ParsedAux.

Invalid function
distance^2
in ParsedAux density.
Syntax error: Unknown identifier"

Here is the input file I've been using:
```
[Mesh]
    [file_mesh]
        type = FileMeshGenerator
        file = meshes/1H_test.msh
    []
    [interior_nodeset]
        type = BoundingBoxNodeSetGenerator
        new_boundary = interior_nodeset
        bottom_left = '-0.01 -0.01 -0.01'
        top_right = '1.01 1.01 1.01'
        input = file_mesh
    []
[]

[Variables]
    [./u]
    [../]
[]

[AuxVariables]
    [./distance]
    [../]
    [./penetration]
    [../]
    [./density]
    [../]
[]

[AuxKernels]
    [./nodal_distance_aux]
        type = NearestNodeDistanceAux
        variable = distance
        boundary = interior_nodeset
        paired_boundary = hydrosurf
    [../]
    [./penetration_aux]
        type = PenetrationAux
        variable = penetration
        boundary = interior_nodeset
        paired_boundary = hydrosurf
    [../]
    [./density]
        type = ParsedAux
        variable = density
        coupled_variable = 'distance'
        expression = 'distance^2'
    [../]
[]

[Problem]
    type = FEProblem
    solve = false
[]

[Executioner]
    type = Steady
[]

[Outputs]
    file_base = densityTest1Hole
    execute_on = 'timestep_end'
    exodus = true
[]
```

I've looked through the examples on the website and I can't find where my syntax is incorrect. Any help would be greatly appreciated!

**Comment by wlindqwister** [comment]
Never mind, I found a small syntax error in ParsedAux that I overlooked. My bad!

**Reply by GiudGiud** [reply]
EDIT: (I'm replying to another poster who seemed to have deleted their message)

You either have a typo in the expression

Or one of the terms in the expression should be in one of the other parameters that describe what type it is

For example a variable should be specified in coupled_variables, a functor in functor_symbols
